src/databricks/labs/ucx/source_code/notebooks/cells.py

```diff
@@ -289,9 +289,11 @@ def requires_isolated_pi(self) -> str:
 
     @classmethod
     def of_language(cls, language: Language) -> CellLanguage:
-        # TODO: Should this not raise a ValueError if the language is not found?
-        #  It also  causes a GeneratorExit exception to be raised. Maybe an explicit loop is better.
-        return next((cl for cl in CellLanguage if cl.language == language))
+        for cell_language in CellLanguage:
+            if cell_language.language == language:
+                return cell_language
+        msg = f"Unsupported cell language: {language}"
+        raise ValueError(msg)
```

Comment:
This would raise `StopIteration` if the language wasn't supported. The TODO has now been resolved: it raises `ValueError` instead.